silverback/runner.py

```diff
@@ -116,42 +161,69 @@ async def run(self):
         else:  # use empty state
             self.state = AppState(last_block_seen=-1, last_block_processed=-1)
 
-        # Initialize broker (run worker startup events)
-        await self.app.broker.startup()
-
         # Execute Silverback startup task before we init the rest
-        if startup_tasks := await asyncio.gather(
-            *(task_def.handler.kiq(self.state) for task_def in self.app.tasks[TaskType.STARTUP])
-        ):
-            results = await asyncio.gather(
-                *(startup_task.wait_result() for startup_task in startup_tasks)
+        startup_taskdata_result = await run_taskiq_task_wait_result(
+            self._create_system_task_kicker(TaskType.SYSTEM_USER_TASKDATA), TaskType.STARTUP
+        )
+
+        if startup_taskdata_result.is_err:
+            raise StartupFailure(startup_taskdata_result.error)
+
+        else:
+            startup_task_handlers = map(
+                self._create_task_kicker, startup_taskdata_result.return_value
             )
 
-            if any(result.is_err for result in results):
+            startup_task_results = await run_taskiq_task_group_wait_results(
+                (task_handler for task_handler in startup_task_handlers), self.state
+            )
+
+            if any(result.is_err for result in startup_task_results):
                 # NOTE: Abort before even starting to run
-                raise StartupFailure(*(result.error for result in results if result.is_err))
+                raise StartupFailure(
+                    *(result.error for result in startup_task_results if result.is_err)
+                )
 
             elif self.recorder:
-                converted_results = map(TaskResult.from_taskiq, results)
+                converted_results = map(TaskResult.from_taskiq, startup_task_results)
                 await asyncio.gather(*(self.recorder.add_result(r) for r in converted_results))
 
             # NOTE: No need to handle results otherwise
 
         # Create our long-running event listeners
+        new_block_taskdata_results = await run_taskiq_task_wait_result(
+            self._create_system_task_kicker(TaskType.SYSTEM_USER_TASKDATA), TaskType.NEW_BLOCK
+        )
```

Comment:
issue: The return type of `self._create_system_task_kicker()` doesn't appear to match what `        new_block_taskdata_results = await run_taskiq_task_wait_result()` wants?

Reply:
Yes should update that, both types will have `.kiq(*args, **kwargs)` method that submits a message to the broker

Reply:
[84f9d11](https://github.com/ApeWorX/silverback/pull/82/commits/84f9d1112c017f4c4d8f4ab4239a48cc83b834e1)